Classes/Controller/DocumentController.php

```diff
@@ -159,22 +159,31 @@ public function discardConfirmAction(\EWW\Dpf\Domain\Model\Document $document)
      */
     public function discardAction(\EWW\Dpf\Domain\Model\Document $document)
     {
-        // remove document from local index
-        $elasticsearchRepository = $this->objectManager->get(ElasticsearchRepository::class);
-        // send document to index
-        $elasticsearchRepository->delete($document, "");
+        try {
+            // remove document from local index
+            $elasticsearchRepository = $this->objectManager->get(ElasticsearchRepository::class);
+            // send document to index
+            $elasticsearchRepository->delete($document, "");
 
-        $this->documentRepository->remove($document);
+            $this->documentRepository->remove($document);
 
-        $key = 'LLL:EXT:dpf/Resources/Private/Language/locallang.xlf:document_discard.success';
+            $key = 'LLL:EXT:dpf/Resources/Private/Language/locallang.xlf:document_discard.success';
+            $severity = \TYPO3\CMS\Core\Messaging\AbstractMessage::OK;
 
-        $args = array();
+        } catch (\Exception $exception) {
 
-        $message = \TYPO3\CMS\Extbase\Utility\LocalizationUtility::translate($key, 'dpf', $args);
-        $message = empty($message) ? "" : $message;
+            $severity = \TYPO3\CMS\Core\Messaging\AbstractMessage::ERROR;
 
-        $this->addFlashMessage($message, '', \TYPO3\CMS\Core\Messaging\AbstractMessage::OK);
+            if ($exception instanceof \EWW\Dpf\Exceptions\ConnectionErrorException) {
```

Comment:
This pattern is repeated at least 10 times for this single exception type (other \EWW\Dpf\Exception have the same code smell).

If you check for the type of exception and use the same language key over and over again why don't you encapsulate that value and let the exception itself return it. You could create an Interface for our custom dpf exceptions e.g. `DPFExceptionInterface`and get the value that way. Custom exceptions implement that interface if they want to and you'd only have to write the value once and ask the exception for it instead of copy&pasting it all the time.

That would reduce duplication and shrink the if/elseif/elseif/… stuff to something short like this:

```
if ($exception instanceof DPFExceptionInterface) {
    $key = $exception->languageKey();
} else {
    $key = "SomeDefaultValueWhatever";
}
```

See: https://stackoverflow.com/a/5579385/842529

Reply:
I followed your suggestion and revised the code accordingly.